From 1fa0bf000b6505d570558bb45f8e07e403feb7c1 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 23 Jun 2019 11:37:27 +0300
Subject: [PATCH] [SW-0] uart: export additional functions

The preboot_puts_hex() and preboot_puts_char() functions appear to be
useful at least during debug.
Export them, so we can use them also outside the uart.c file.

Change-Id: I1b0a09e550ec05b9c1df60adbeab802795f5effa
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/uart.c b/common/uart.c
index 3f57dc1..1a26a85 100644
--- a/common/uart.c
+++ b/common/uart.c
@@ -83,14 +83,14 @@
 	preboot_puts("\n");
 }
 
-static inline void preboot_puts_hex(const char *log, unsigned int val, bool pad)
+void preboot_puts_hex(const char *log, unsigned int val, bool pad)
 {
 	preboot_puts_raw(log);
 	preboot_puts_raw("0x");
 	preboot_puth(val, pad);
 }
 
-static inline void preboot_puts_char(const char *str, unsigned char val)
+void preboot_puts_char(const char *str, unsigned char val)
 {
 	preboot_puts_hex(str, val, false);
 }
diff --git a/common/uart.h b/common/uart.h
index 00712c9..595ff10 100644
--- a/common/uart.h
+++ b/common/uart.h
@@ -17,5 +17,7 @@
 int preboot_uint2hexstring(char *str, unsigned int num, bool pad);
 void preboot_puth_raw(unsigned int val, bool pad);
 void preboot_puth(unsigned int val, bool pad);
+void preboot_puts_hex(const char *log, unsigned int val, bool pad);
+void preboot_puts_char(const char *str, unsigned char val);
 
 #endif /* _HL_UART_H_ */
